x-pack/plugins/watcher/server/plugin.ts

```diff
@@ -45,21 +45,26 @@ export class WatcherServerPlugin implements Plugin<void, void, any, any> {
     this.log = ctx.logger.get();
   }
 
-  async setup(
-    { http, elasticsearch: elasticsearchService }: CoreSetup,
-    { licensing }: Dependencies
-  ) {
+  async setup({ http, getStartServices }: CoreSetup, { licensing }: Dependencies) {
     const router = http.createRouter();
     const routeDependencies: RouteDependencies = {
       router,
       getLicenseStatus: () => this.licenseStatus,
     };
 
-    const config = { plugins: [elasticsearchJsPlugin] };
-    const watcherESClient = elasticsearchService.createClient('watcher', config);
+    const getWatcherEsClient = async () => {
+      const coreStart = (await getStartServices())[0];
+      const config = { plugins: [elasticsearchJsPlugin] };
+      return coreStart.elasticsearch.createClient('watcher', config);
+    };
     http.registerRouteHandlerContext('watcher', (ctx, request) => {
       return {
-        client: watcherESClient.asScoped(request),
+        client: {
```

Comment:
This looks more like a hack than a solution to me. Snapshot and Restore app has a very similar logic than watcher to create the ES client. Should we move all the logic of the client creation + registering routes directly in the `start()` method?